src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,35 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> map = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (map.containsKey(userId)) {
+            return getUser(userId,executor);
+
+        }
+        try {
+            Thread.sleep(1000);
```

Comment:
Using 'Thread.sleep' in an asynchronous context is not advisable as it blocks the current thread. Consider refactoring this to use a non-blocking approach.